client/app.jsx

```diff
@@ -55,6 +57,39 @@ export default class App extends React.Component {
             }
           });
         });
+      fetch(`/api/workout/get-workout/${user.userId}`)
+        .then(res => res.json())
+        .then(result => {
+          const tempWorkout = [];
+          let day = {};
+          let exercise = [];
+          let dayName = '';
+          for (let i = 0; i < result.length; i++) {
+            if (dayName !== result[i].workoutDayName) {
+              if (exercise.length > 0) {
+                day = {
+                  workoutName: dayName,
+                  exercise: exercise
+                };
+                tempWorkout.push(day);
+                day = {};
+                exercise = [];
+              }
+              dayName = result[i].workoutDayName;
+            }
+            if (dayName === result[i].workoutDayName) {
+              exercise.push(result[i]);
+            }
+            if (i === result.length - 1) {
+              day = {
+                workoutName: dayName,
+                exercise: exercise
+              };
+              tempWorkout.push(day);
+            }
+          }
```

Comment:
It is best to have the server format the data in a way that the client can use it without doing stuff like this. The servers job is to server the data how the client needs it. Plus it is ideal to do data processing on the server because you never know what type of device the client will be using, data processing can be very slow on older less powerful devices.